tests/CMakeLists.txt

```diff
@@ -30,42 +33,250 @@ set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+  ${CMAKE_SOURCE_DIR}/src/ssh
+  ${CMAKE_SOURCE_DIR}/src/sshfs_mount
+)
+
+# Base include directories needed by all test components
+set(BASE_INCLUDE_DIRS
+  ${CMAKE_SOURCE_DIR}
+  ${CMAKE_SOURCE_DIR}/src
+  ${CMAKE_SOURCE_DIR}/include # For multipass headers
+  ${CMAKE_SOURCE_DIR}/3rd-party/fmt/include # For fmt headers
+  ${CMAKE_SOURCE_DIR}/3rd-party/scope_guard # For scope_guard.hpp
+)
+
+# Settings test object library
+add_library(test_settings_obj OBJECT
+  test_global_settings_handlers.cpp
+  test_instance_settings_handler.cpp
+  test_persistent_settings_handler.cpp
+  test_remote_settings_handler.cpp
+  test_setting_specs.cpp
+  test_settings.cpp
+)
+target_include_directories(test_settings_obj PRIVATE
+  ${BASE_INCLUDE_DIRS}
+  ${TEST_COMMON_INCLUDE_PATHS}
+  ${CMAKE_SOURCE_DIR}/src/settings
+)
+target_link_libraries(test_settings_obj PRIVATE
+  ${TEST_COMMON_DEPS}
+  settings
+)
+
+# Network and parser object library for tests needing protobuf
+add_library(test_network_parser_obj OBJECT
+  test_argparser.cpp
+)
+target_include_directories(test_network_parser_obj PRIVATE
+  ${BASE_INCLUDE_DIRS}
+  ${TEST_COMMON_INCLUDE_PATHS}
+  ${CMAKE_BINARY_DIR}
+  ${CMAKE_SOURCE_DIR}/3rd-party/yaml-cpp/include
+)
+target_link_libraries(test_network_parser_obj PRIVATE
+  ${TEST_COMMON_DEPS}
+  protobuf::libprotobuf
+  client
+  daemon
+)
+
+# Core object libraries
+add_library(test_minimal_obj OBJECT
   common.cpp
-  daemon_test_fixture.cpp
+  main.cpp
+  mock_logger.cpp
+  mock_standard_paths.cpp
+  temp_dir.cpp
+  temp_file.cpp
+)
+target_include_directories(test_minimal_obj PRIVATE
+  ${BASE_INCLUDE_DIRS}
+  ${TEST_COMMON_INCLUDE_PATHS}
+)
+target_link_libraries(test_minimal_obj PRIVATE ${TEST_COMMON_DEPS})
+
+add_library(test_fs_obj OBJECT
   file_operations.cpp
-  image_host_remote_count.cpp
+  path.cpp
+)
+target_include_directories(test_fs_obj PRIVATE
+  ${BASE_INCLUDE_DIRS}
+  ${TEST_COMMON_INCLUDE_PATHS}
+  ${CMAKE_SOURCE_DIR}/3rd-party/yaml-cpp/include
+)
+target_link_libraries(test_fs_obj PRIVATE ${TEST_COMMON_DEPS})
+
+add_library(test_json_obj OBJECT
   json_test_utils.cpp
-  main.cpp
+  blueprint_test_lambdas.cpp
+)
+target_include_directories(test_json_obj PRIVATE
+  ${BASE_INCLUDE_DIRS}
+  ${TEST_COMMON_INCLUDE_PATHS}
+  ${CMAKE_SOURCE_DIR}/3rd-party/yaml-cpp/include
+  ${CMAKE_BINARY_DIR}
+)
+target_link_libraries(test_json_obj PRIVATE ${TEST_COMMON_DEPS} protobuf::libprotobuf)
+
+add_library(test_network_obj OBJECT
   mischievous_url_downloader.cpp
-  mock_logger.cpp
+  image_host_remote_count.cpp
+)
+target_include_directories(test_network_obj PRIVATE
+  ${BASE_INCLUDE_DIRS}
+  ${TEST_COMMON_INCLUDE_PATHS}
+  ${CMAKE_SOURCE_DIR}/src/network
+)
+target_link_libraries(test_network_obj PRIVATE ${TEST_COMMON_DEPS})
+
+add_library(test_daemon_obj OBJECT
+  daemon_test_fixture.cpp
+)
+target_include_directories(test_daemon_obj PRIVATE
+  ${BASE_INCLUDE_DIRS}
+  ${TEST_COMMON_INCLUDE_PATHS}
+  ${CMAKE_SOURCE_DIR}/src/daemon
```

Comment:
Listing includes paths per-target seems a little bit verbose to me. Because the includes will only hurt the build performance when they're actually included, we can just list all the include paths at the top:

```cmake
include_directories(/*list of all include paths used by the tests*/)
```

The downside would be that all the targets would be able to use any headers listed in the include path, which IMO is reasonable given that we don't have to micro-manage the include paths per target. What do you think?